test/jdk/sun/net/www/protocol/http/AuthCache.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8304818
+ * @modules java.base/sun.net.www.protocol.http
+ * @library /test/lib
+ * @build jdk.test.lib.util.ForceGC
+ * @run main/othervm AuthCache
+ */
+
+import com.sun.net.httpserver.BasicAuthenticator;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import java.io.IOException;
+import java.lang.ref.PhantomReference;
+import java.net.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+import sun.net.www.protocol.http.AuthCacheValue;
+import jdk.test.lib.util.ForceGC;
+
+public class AuthCache {
+    static class ClientAuth extends Authenticator {
+        private final String realm;
+        private final String username;
+        private final String password;
+        private AtomicBoolean wasCalled = new AtomicBoolean();
```

Comment:
`wasCalled` is accessed only from `synchronized` methods. We can use plain `boolean` instead